libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelper.java

```diff
@@ -526,7 +526,7 @@ private static int findAnnotationIndex(CurrentLegAnnotation currentLegAnnotation
     for (int i = distanceIndex; i < legDistances.size(); i++) {
       Double distance = legDistances.get(i);
       annotationDistancesTraveled += distance;
-      if (annotationDistancesTraveled > distanceTraveled) {
+      if (annotationDistancesTraveled > distanceTraveled || i == legDistances.size() -1 ) {
```

Comment:
What is here fixed? It's possible that `return INDEX_ZERO` in `line 535` not reachable anymore?

Is that covered by a test case?

Reply:
Correct, `INDEX_ZERO` should not be returned anymore. We left it untouched for historic reasons. 

`distanceTraveled >= annotationDistancesTraveled` can happen if you are at the end of leg. Before we returned index zero, which is wrong. Index 0 is the start of leg. If you have travelled the whole leg, we need to return the last index.

We actually found this bug when writing unit tests, so this is covered in a unit test implicitly.

Reply:
Nice. Sounds good for ne 💪